### ISSUE TITLE
Installed tool shed repositories usability enhancements

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#1268]
- [ ] allow sorting on all columns
- [ ] add a column for "date installed"
- [ ] group different versions of the same tool together
- [ ] hide packages and suite by default (maybe separate tab?)
- [ ] get an update all button (@erasche and others have wanted this since GCC2014)
- [x] add/remove tools without restarting the server, leveraging existing "tool reload" infrastructure probably.
- [ ] add/remove admin without restarting the server (last time I checked this was not possible if i'm not mistaken)
- [ ] Add the "non-dropdown action" of the Name button to the list of actions on the dropdown.

**Comment by bgruening** [comment]
Added a few more items.

**Comment by yhoogstrate** [comment]
add/remove admin without restarting the server (last time I checked this was not possible if i'm not mistaken)

**Comment by hexylena** [comment]
Done @yhoogstrate

**Comment by yhoogstrate** [comment]
I am not sure if this is a bug or a feature. When you have no data manager installed (clean galaxy instance), the Data Manager menu is kind of hidden. This prevents the admin from reloading the built-in *.loc files from e.g. `all_fasta.loc`.

![screenshot from 2015-12-10 22 14 49](https://cloud.githubusercontent.com/assets/2264697/11728738/cccdda96-9f8b-11e5-828c-835b6b01a10b.png)

**Comment by peterjc** [comment]
@yhoogstrate you said it was kind of hidden, but it is possible to reload built in `*.loc` files from the _Admin_ menus? How - we'd like to use that functionality.

**Comment by peterjc** [comment]
Another wish list entry for the installed tools page:
- Run tool tests

See also #3136 and #3137

**Comment by nsoranzo** [comment]
@peterjc It is possible to reload **data tables**: in the Admin page click "Local data", then select a data table from the list "View Tool Data Table Entries" and finally click the reload icon.

**Comment by peterjc** [comment]
@nsoranzo sadly no, I get the same screen shot as @yhoogstrate posted - there is no list of tables to click on. Should we open this as a new bug?

**Comment by nsoranzo** [comment]
@peterjc :open_mouth: I wasn't aware! Yes, please, open an issue for this.

**Comment by eschen42** [comment]
@erasche Could you please add the following item? Alternatively, I could open a new issue.

TL;DR; **Add the "non-dropdown action" of the `Name` button to the list of actions on the dropdown.**

On the `Admin:Installed tool shed repositories` page, the `Name` buttons appear with a dropdown.  The dropdown lists only two choices, and neither of those choices are what happens when you click the non-dropdown portion of the button.

I am so used to the convention that a dropdown lets you choose several actions and clicking the button chooses the default from among them that it has taken me two years to discover that the non-dropdown action (installed toolshed repository) is different from the dropdown choices (update toolshed status; deactivate or uninstall)!

So, for better UX, please add the `installed toolshed repository` choice to the dropdown.  I don't know whether that is the best label for the action; perhaps "review installed repository" would be more informative.
